azurelinuxagent/ga/exthandlers.py

```diff
@@ -1084,29 +1090,63 @@ def get_status_file_path(self, extension=None):
+            add_periodic(delta=logger.EVERY_HALF_HOUR, name=ext.name, version=self.ext_handler.properties.version,
+                         op=WALAEventOperation.StatusProcessing, is_success=False, message=ustr(e),
+                         log_event=False)
+
+            ext_status.message = ustr(e)
+            ext_status.status = VALID_EXTENSION_STATUS.error
+            ext_status.code = ExtensionErrorCodes.PluginSettingsStatusInvalid if \
+                e.code == ExtensionStatusError.InvalidJsonFile else \
+                ExtensionErrorCodes.PluginUnknownFailure
+
+            return ext_status
+
+        # We did not encounter InvalidJsonFile/StatusFileNotFound and thus the status file was correctly written and has
+        # valid json.
         try:
-            data_str = fileutil.read_file(ext_status_file)
-            data = json.loads(data_str)
             parse_ext_status(ext_status, data)
-        except IOError as e:
-            ext_status.message = u"Failed to get status file {0}".format(e)
-            ext_status.code = -1
-            ext_status.status = "error"
-        except ExtensionError as e:
-            ext_status.message = u"Malformed status file {0}".format(e)
-            ext_status.code = ExtensionErrorCodes.PluginSettingsStatusInvalid
-            ext_status.status = "error"
-        except ValueError as e:
-            ext_status.message = u"Malformed status file {0}".format(e)
-            ext_status.code = -1
-            ext_status.status = "error"
+            if len(data_str) > _MAX_STATUS_FILE_SIZE_IN_BYTES:
+                raise ExtensionStatusError(msg="For Extension Handler {0}-{1} for the sequence number {2}, the status "
+                                               "file {3} of size {4} bytes is too big. Max Limit allowed is {5} bytes"
+                                           .format(ext.name, self.ext_handler.properties.version, seq_no,
+                                                   ext_status_file, len(data_str), _MAX_STATUS_FILE_SIZE_IN_BYTES),
+                                           code=ExtensionStatusError.MaxSizeExceeded)
+        except ExtensionStatusError as e:
+            msg = u"For Extension Handler {0}-{1} for the sequence number {2}, the status file {3}. " \
+                  u"Encountered the following error: {4}".format(ext.name, self.ext_handler.properties.version, seq_no,
+                                                                 ext_status_file, ustr(e))
+            logger.periodic_warn(logger.EVERY_DAY, u"[PERIODIC] " + msg)
+            add_periodic(delta=logger.EVERY_HALF_HOUR, name=ext.name, version=self.ext_handler.properties.version,
+                         op=WALAEventOperation.StatusProcessing, is_success=False, message=msg, log_event=False)
+
+            if e.code == ExtensionStatusError.MaxSizeExceeded:
+                ext_status.message = self._truncate_message(ext_status.message)
```

Comment:
we discussed how truncating from > 128k to 200 may not be the best approach. And also the loop below wouldn't catch the case where there are many, many, substatuses. Why not truncating each field (message and then each substatus) to 128K and keep appending until you exceed 128K?

Reply:
Done.